presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -599,6 +602,27 @@ private QueryInfo buildQueryInfo(SqlQueryScheduler scheduler)
         return queryInfo;
     }
 
+    @Override
+    public boolean shouldWaitForMinWorkers()
+    {
+        return shouldWaitForMinWorkers(analysis.getStatement());
+    }
+
+    private boolean shouldWaitForMinWorkers(Statement statement)
+    {
+        if (statement instanceof Query) {
```

Comment:
Also, DDL statements should not wait. We can classify these using `QueryType` via `StatementUtils`. Waiting can always be skipped for `DATA_DEFINITION`, `DESCRIBE`, `EXPLAIN`.

Reply:
Actually, I don't think this is a problems, since `SqlQueryExecution` does not happen for DDL, and by the time we get here, `EXPLAIN` has already been rewritten (at least I think so ... double check this).

Reply:
@electrum is right. There's also a test case `testInsufficientWorkerNodesInternalSystemQuery` which checks that we don't wait for `EXPLAIN` query.